compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_posix_libcdep.cpp

```diff
@@ -38,10 +38,36 @@
 // because we do not require a C++ ABI library to be linked to a program
 // using sanitizers; if it's not present, we'll just use the mangled name.
 namespace __cxxabiv1 {
+#  if !SANITIZER_AIX
 extern "C" SANITIZER_WEAK_ATTRIBUTE char *__cxa_demangle(const char *mangled,
                                                          char *buffer,
                                                          size_t *length,
                                                          int *status);
+#  else
+// `weak` attribute without definition on AIX will cause linking time undefined
+// symbol error, we use dlopen/dlsym here to find related symbol.
+typedef char *__cxa_demangle_t(const char *mangled, char *buffer,
+                               size_t *length, int *status);
+
+char *__cxa_demangle(const char *mangled, char *buffer, size_t *length,
```

Comment:
The sanitizer _should not_ define a `__cxa_demangle` that could satisfy a link dependency from the user program.

Reply:
Consider an approach that changes the linkage name used on AIX for calls to the function in the sanitizer runtime.